build_tools/third_party/nrfx/CMakeLists.txt

```diff
@@ -20,11 +20,17 @@ target_sources(nrfx_nrf52840
 target_include_directories(nrfx_nrf52840
   PUBLIC
     ${CMSIS_5_SOURCE_DIR}/CMSIS/Core/Include/
-  PRIVATE
+    ${nrfx_DIR}
+    ${nrfx_DIR}/drivers/include/
+    ${nrfx_DIR}/hal/
+    ${nrfx_DIR}/templates/
     ${nrfx_DIR}/mdk/
 )
 
 target_compile_definitions(nrfx_nrf52840
-  PRIVATE
+  PUBLIC
```

Comment:
Do we really need this as a public compile definition? Do we only do this here to pass it to `nrfx_uart.c`? In that case I would propose a different solution, especially because we should move `nrfx_uart.c` (see my other comment).

Reply:
Yeash, this is needed.